### PR TITLE
[WB-3606] pyagent run stopping fix

### PR DESCRIPTION
**Pull request by farizrahman4u**

### CODE REVIEW

File: wandb/agents/pyagent.py [hunk 1]
```diff
@@ -27,6 +27,9 @@
 def _terminate_thread(thread):
     if not thread.is_alive():
         return
+    if hasattr(thread, "_terminated"):
+        return
+    thread._terminated = True
```
Comment (vanpelt) [inline wandb/agents/pyagent.py:32]: So this is to catch the time between stopping a thread and it actually finishing execution?

**Review by vanpelt — APPROVED** [review]
Looks good, it would be nice to add a test like those in `wandb_integration_test.py` that ran a little sweep with some arbitrary sleeps in the `train` function.  Likely outside of the scope of this PR, but should be fairly easy to implement.

File: wandb/agents/pyagent.py (hunk 1 above)
Comment (farizrahman4u) [reply wandb/agents/pyagent.py:32]: Yes.

**Review by raubitsj — APPROVED** [review]
approved.

I did notice a change in how this handles stopped runs, filed a bug to look at this:
https://wandb.atlassian.net/browse/WB-3793

**Pull request merged**